incorrect error message when creating KV watches

## Defect

 - [ ] Included nats.go version
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:

`nats-server: v2.9.19`
`github.com/nats-io/nats.go v1.27.1`

#### OS/Container environment:

MacOS 13.4.1
`go version go1.20.5 darwin/arm64`

#### Steps or code to reproduce the issue:

Start a JetStream enabled NATS server:

```
nats-server -js
```

Spawn a watch on an invalid key:

```go
package main

import (
	"github.com/nats-io/nats.go"
)

func main() {
	nc, _ := nats.Connect(nats.DefaultURL)

	js, err := nc.JetStream()
	if err != nil {
		panic(err)
	}

	kv, err := js.KeyValue("bucket")
	if err != nil {
		panic(err)
	}

	_, err = kv.Watch("test..")
	if err != nil {
		panic(err)
	}
}
```

#### Expected result:

An error message similar to the one the CLI provides:

```
$ nats kv watch bucket test..
nats: error: nats: consumer filter subject is not a valid subset of the interest subjects
```

#### Actual result:

```
panic: nats: jetstream not enabled
```

--- a/jetstream/kv.go
+++ b/jetstream/kv.go
@@ -906,14 +906,14 @@ func bucketValid(bucket string) bool {
 }
 
 func keyValid(key string) bool {
-	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' {
+	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' || strings.Contains(key, "..") {
 		return false
 	}
 	return validKeyRe.MatchString(key)
 }
 
 func searchKeyValid(key string) bool {
-	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' {
+	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' || strings.Contains(key, "..") {
 		return false
 	}
 	return validSearchKeyRe.MatchString(key)

--- a/jetstream/test/kv_test.go
+++ b/jetstream/test/kv_test.go
@@ -620,6 +620,13 @@ func TestKeyValueWatch(t *testing.T) {
 		_, err = kv.Watch(ctx, "foo.")
 		expectErr(t, err, jetstream.ErrInvalidKey)
 
+		// consecutive dots (empty token) must be rejected early, not passed to server
+		_, err = kv.Watch(ctx, "foo..bar")
+		expectErr(t, err, jetstream.ErrInvalidKey)
+
+		_, err = kv.Watch(ctx, "test..")
+		expectErr(t, err, jetstream.ErrInvalidKey)
+
 		// conflicting options
 		_, err = kv.Watch(ctx, "foo", jetstream.IncludeHistory(), jetstream.UpdatesOnly())
 		expectErr(t, err, jetstream.ErrInvalidOption)

--- a/jetstream/test/kv_test.go
+++ b/jetstream/test/kv_test.go
@@ -624,7 +624,7 @@ func TestKeyValueWatch(t *testing.T) {
 		_, err = kv.Watch(ctx, "foo..bar")
 		expectErr(t, err, jetstream.ErrInvalidKey)
 
-		_, err = kv.Watch(ctx, "test..")
+		_, err = kv.Watch(ctx, "a..b.c")
 		expectErr(t, err, jetstream.ErrInvalidKey)
 
 		// conflicting options

--- a/kv.go
+++ b/kv.go
@@ -577,14 +577,14 @@ func bucketValid(bucket string) bool {
 }
 
 func keyValid(key string) bool {
-	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' {
+	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' || strings.Contains(key, "..") {
 		return false
 	}
 	return validKeyRe.MatchString(key)
 }
 
 func searchKeyValid(key string) bool {
-	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' {
+	if len(key) == 0 || key[0] == '.' || key[len(key)-1] == '.' || strings.Contains(key, "..") {
 		return false
 	}
 	return validSearchKeyRe.MatchString(key)

--- a/test/kv_test.go
+++ b/test/kv_test.go
@@ -413,6 +413,9 @@ func TestKeyValueWatch(t *testing.T) {
 
 		_, err = kv.Watch("foo.")
 		expectErr(t, err, nats.ErrInvalidKey)
+
+		_, err = kv.Watch("foo..bar")
+		expectErr(t, err, nats.ErrInvalidKey)
 	})
 
 	t.Run("filtered watch with no filters", func(t *testing.T) {

